File: cpp/42c_1607A_LinearKeyboard.cpp
#include<bits/stdc++.h>

int main(){
    int t, cnt;
    std::vector<int> ans;
    std::string s, k;
    std::cin>>t;
    for (int i = 0; i < t; i++){
        std::cin>>k;
        std::cin>>s;
        std::unordered_map<char,int> hmap;
        for(int j = 0; j < k.size(); j++){
            hmap[k[j]] = j;
        }
        cnt = 0;
        for(int j = 1; j < s.size(); j++){
            cnt += abs((hmap[s[j]] - hmap[s[j-1]]));
        }
        ans.push_back(cnt);
    }
    for (auto i : ans){
        std::cout<<i<<"\n";
    }
    return 0;
}

File: python/42c_1607A_LinearKeyboard.py
def main():
    ans = []
    t = int(input())
    for _ in range(t):
        k = input().strip()
        s = input().strip()
        hmap = {}
        for i in range(len(k)):
            hmap[k[i]] = i
        cnt = 0
        for i in range(1,len(s)):
            cnt += abs(hmap[s[i]] - hmap[s[i-1]])
        ans.append(cnt)
    for i in ans:
        print(i)

if __name__ == "__main__":
    main()